test/Microsoft.Azure.Devices.Edge.Test/PriorityQueues.cs

```diff
@@ -0,0 +1,135 @@
+                                   ["ActualSource"] = "relayerModule.receive",
+                                   ["TestDescription"] = "unnecessary"
+                               }
+                           }
+                       });
+                    builder.AddModule(loadGenModuleName, loadGenImage)
+                        .WithEnvironment(new[]
+                        {
+                            ("testResultCoordinatorUrl", trcUrl),
+                            ("senderType", "PriorityMessageSender"),
+                            ("trackingId", "e2eTestTrackingId"),
+                            ("testDuration", "00:00:20"),
+                            ("messageFrequency", "00:00:01")
+                        });
+
+                    builder.GetModule(ModuleName.EdgeHub)
+                        .WithDesiredProperties(new Dictionary<string, object>
+                        {
+                            ["routes"] = new Dictionary<string, object>
+                            {
+                                ["LoadGenToRelayer1"] = new Dictionary<string, object>
+                                {
+                                    ["route"] = "FROM /messages/modules/" + loadGenModuleName + "/outputs/pri0 INTO BrokeredEndpoint('/modules/" + relayerModuleName + "/inputs/input1')",
+                                    ["priority"] = 0
+                                },
+                                ["LoadGenToRelayer2"] = new Dictionary<string, object>
+                                {
+                                    ["route"] = "FROM /messages/modules/" + loadGenModuleName + "/outputs/pri1 INTO BrokeredEndpoint('/modules/" + relayerModuleName + "/inputs/input1')",
+                                    ["priority"] = 1
+                                },
+                                ["LoadGenToRelayer3"] = new Dictionary<string, object>
+                                {
+                                    ["route"] = "FROM /messages/modules/" + loadGenModuleName + "/outputs/pri2 INTO BrokeredEndpoint('/modules/" + relayerModuleName + "/inputs/input1')",
+                                    ["priority"] = 2
+                                },
+                                ["LoadGenToRelayer4"] = new Dictionary<string, object>
+                                {
+                                    ["route"] = "FROM /messages/modules/" + loadGenModuleName + "/outputs/pri3 INTO BrokeredEndpoint('/modules/" + relayerModuleName + "/inputs/input1')",
+                                    ["priority"] = 3
+                                }
+                            }
+                        });
+                });
+
+            EdgeDeployment deployment = await this.runtime.DeployConfigurationAsync(addInitialConfig, token);
+
+            // Wait for loadGen to send some messages
+            await Task.Delay(TimeSpan.FromSeconds(30));
```

Comment:
Delay will be different in different platforms, is there any better approach to check deployment state?

Reply:
How can you ensure that within 30 secs, all given priorities will be used at least one to send messages?

Reply:
DeployConfigurationAsync waits until all modules are up and running. Therefore, we know that loadGen is sending messages when we have this delay. The delay is a bit longer than the testDuration of LoadGen, so we know that LoadGen will be finished sending messages. We wait a bit of extra time for loadGen to send messages to the TRC. I think this should work on all platforms. If not, we can bump up the last delay time, which seems to be the only place where the bottleneck could be.

Reply:
To your second comment, we can't ensure that all priorities will be used at least once, but most probably they will be since they're chosen randomly. We can bump up the number of messages to be almost certain. 

But it's also not even essential that every route is used. The test will still pass, and the behavior will be correct.

Reply:
Since you mentioned the relationship between "testDuration" and this delay value; please derive this delay value based on "testDuration".  For example. TimeSpan.Parse(testDuration).Add(TimeSpan.FromSeconds(10)).

Reply:
nit:To be safe, you can throw an exception in PriorityMessageSender when only 1 priority is used.